src/components/ExpensifyCashLogo.js

```diff
@@ -3,26 +3,41 @@ import PropTypes from 'prop-types';
 import ProductionLogo from '../../assets/images/expensify-cash.svg';
 import DevLogo from '../../assets/images/expensify-cash-dev.svg';
 import StagingLogo from '../../assets/images/expensify-cash-stg.svg';
-import CONFIG from '../CONFIG';
 import CONST from '../CONST';
+import Environment from '../libs/Environment/index';
 
 const propTypes = {
     width: PropTypes.number.isRequired,
     height: PropTypes.number.isRequired,
 };
 
-const ExpensifyCashLogo = (props) => {
-    switch (CONFIG.EXPENSIFY.ENVIRONMENT) {
-        case CONST.ENVIRONMENT.PRODUCTION:
-            return <ProductionLogo width={props.width} height={props.height} />;
-        case CONST.ENVIRONMENT.STAGING:
-            return <StagingLogo width={props.width} height={props.height} />;
-        default:
-            return <DevLogo width={props.width} height={props.height} />;
+class ExpensifyCashLogo extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            environment: CONST.ENVIRONMENT.PRODUCTION,
+        };
     }
-};
 
-ExpensifyCashLogo.propTypes = propTypes;
-ExpensifyCashLogo.displayName = 'ExpensifyCashLogo';
+    componentDidMount() {
+        Environment.getEnvironment()
+            .then((environment) => {
+                this.setState({environment});
+            });
+    }
 
+    render() {
+        switch (this.state.environment) {
+            case CONST.ENVIRONMENT.PRODUCTION:
+                return <ProductionLogo width={this.props.width} height={this.props.height} />;
```

Comment:
NAB, these could be DRYed up a little bit more too. Would need to do something like:

```
let logoComponent;
switch (this.state.environment) {
    case CONST.ENVIRONMENT.PRODUCTION:
        logoComponent = ProductionLogo;
        break;
}
return (<logoComponent width={this.props.width} height={this.props.height} />);
```
or
```
const logoComponents = {
    [CONST.ENVIRONMENT.PRODUCTION]: ProductionLogo,
};
return (<logoComponents[this.state.environment] width={this.props.width} height={this.props.height} />);
```

Reply:
Cool recommendation, I wasn't aware of that as an option but I like the simplification